packages/doenetml/src/Viewer/renderers/booleanInput.tsx

```diff
@@ -58,20 +63,20 @@ export default React.memo(function BooleanInput(props) {
 
     let valueWhenSetState = useRef(null);
 
-    let inputJXG = useRef(null);
-    let anchorPointJXG = useRef(null);
-    let anchorRel = useRef(null);
+    let inputJXG = useRef<JXGObject | null>(null);
+    let anchorPointJXG = useRef<JXGObject | null>(null);
+    let anchorRel = useRef<[string, string] | null>(null);
 
     const board = useContext(BoardContext);
 
-    let pointerAtDown = useRef(false);
-    let pointAtDown = useRef(false);
+    let pointerAtDown = useRef<[number, number] | null>(null);
+    let pointAtDown = useRef<[number, number] | null>(null);
```

Comment:
`scrCoords` (and hence pointAtDown) is actually a `[number, number, number]`

Reply:
JSXGraph uses [3D homogeneous coordinates](https://en.wikipedia.org/wiki/Homogeneous_coordinates) to represent a point in the plane.

Reply:
Is this also true of `pointerAtDown`?

Reply:
No, that doesn't come from JSXGraph.